src/core/algorithms/association_rules/ar_verifier/ar_verifier.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include "algorithms/algorithm.h"
+#include "algorithms/association_rules/ar.h"
+#include "algorithms/association_rules/ar_algorithm_enums.h"
+#include "algorithms/association_rules/ar_verifier/ar_stats_calculator.h"
+#include "config/tabular_data/input_table_type.h"
+#include "model/table/column_layout_relation_data.h"
+
+namespace algos::ar_verifier {
+class ARVerifier final : public Algorithm {
+private:
+    /* input options */
+    config::InputTable input_table_;
+    InputFormat input_format_ = InputFormat::singular;
+    unsigned int tid_column_index_;
+    unsigned int item_column_index_;
+    bool first_column_tid_;
+
+    std::shared_ptr<model::TransactionalData> transactional_data_;
+    std::list<std::string> string_rule_left_;
+    std::list<std::string> string_rule_right_;
```

Comment:
Why do we want to use a `std::list<>` here?

Reply:
Can we get rid of using these lists? I think it would be better to replace them with a vector both in the existing AR code (if it exists) and in the new code.